bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -243,7 +252,16 @@ public void updateState(String group, String channelId, State value) {
     @Override
     public void dispose() {
         if (webSocketPollingJob != null) {
-            webSocketPollingJob.cancel(true);
+            if (!webSocketPollingJob.isCancelled() && webSocketPollingJob != null) {
+                webSocketPollingJob.cancel(true);
+            }
+            webSocketPollingJob = null;
```

Comment:
Put this in a method. It's copy-pasted all over the place.

Reply:
solved